Build broken with gcc 10.1

I know, no opening of issues for build related stuff when Travis is green, but I feel like this is an exception :)

PSA: If your distro offers an upgrade of gcc to 10.1, don't do it. It broke my ability to build Serenity -_-

Upgraded gcc to 10.1.1 on Fedora 32 today, build failing (log below). Downgraded to 10.0.1, all fine again.

Should be relatively easy to fix, but maybe the compiler is actually emitting a useful warning here?

```
make[2]: Leaving directory '/home/linus/Dev/git/serenity/Libraries/LibJS'
make[2]: Entering directory '/home/linus/Dev/git/serenity/Libraries/LibDesktop'
make -C ../../Services/LaunchServer/ LaunchServerEndpoint.h
make[3]: Entering directory '/home/linus/Dev/git/serenity/Services/LaunchServer'
make[4]: Entering directory '/home/linus/Dev/git/serenity/DevTools/IPCCompiler'
IPCCompiler: C++ main.host.o
In file included from /home/linus/Dev/git/serenity/AK/Assertions.h:36,
                 from /home/linus/Dev/git/serenity/AK/StringView.h:29,
                 from /home/linus/Dev/git/serenity/AK/LogStream.h:35,
                 from /home/linus/Dev/git/serenity/AK/ByteBuffer.h:29,
                 from /home/linus/Dev/git/serenity/AK/BufferStream.h:29,
                 from main.cpp:27:
/home/linus/Dev/git/serenity/AK/Vector.h: In instantiation of ‘T& AK::Vector<T, inline_capacity>::at(size_t) [with T = Endpoint; long unsigned int inline_capacity = 0; size_t = long unsigned int]’:
/home/linus/Dev/git/serenity/AK/Vector.h:287:26:   required from ‘T& AK::Vector<T, inline_capacity>::last() [with T = Endpoint; long unsigned int inline_capacity = 0]’
main.cpp:190:24:   required from here
/home/linus/Dev/git/serenity/AK/Vector.h:276:18: error: comparison of unsigned expression in ‘>= 0’ is always true [-Werror=type-limits]
  276 |         ASSERT(i >= 0 && i < m_size);
      |                ~~^~~~
/home/linus/Dev/git/serenity/AK/Vector.h:276:9: note: in expansion of macro ‘ASSERT’
  276 |         ASSERT(i >= 0 && i < m_size);
      |         ^~~~~~
/home/linus/Dev/git/serenity/AK/Vector.h: In instantiation of ‘const T& AK::Vector<T, inline_capacity>::at(size_t) const [with T = Parameter; long unsigned int inline_capacity = 0; size_t = long unsigned int]’:
/home/linus/Dev/git/serenity/AK/Vector.h:280:66:   required from ‘const T& AK::Vector<T, inline_capacity>::operator[](size_t) const [with T = Parameter; long unsigned int inline_capacity = 0; size_t = long unsigned int]’
main.cpp:269:47:   required from here
/home/linus/Dev/git/serenity/AK/Vector.h:271:18: error: comparison of unsigned expression in ‘>= 0’ is always true [-Werror=type-limits]
  271 |         ASSERT(i >= 0 && i < m_size);
      |                ~~^~~~
/home/linus/Dev/git/serenity/AK/Vector.h:271:9: note: in expansion of macro ‘ASSERT’
  271 |         ASSERT(i >= 0 && i < m_size);
      |         ^~~~~~
/home/linus/Dev/git/serenity/AK/Vector.h: In instantiation of ‘T& AK::Vector<T, inline_capacity>::at(size_t) [with T = Message; long unsigned int inline_capacity = 0; size_t = long unsigned int]’:
/home/linus/Dev/git/serenity/AK/Vector.h:281:54:   required from ‘T& AK::Vector<T, inline_capacity>::operator[](size_t) [with T = Message; long unsigned int inline_capacity = 0; size_t = long unsigned int]’
/home/linus/Dev/git/serenity/AK/Vector.h:74:47:   required from ‘ElementType& AK::VectorIterator<VectorType, ElementType>::operator*() [with VectorType = AK::Vector<Message>; ElementType = Message]’
main.cpp:247:39:   required from here
/home/linus/Dev/git/serenity/AK/Vector.h:276:18: error: comparison of unsigned expression in ‘>= 0’ is always true [-Werror=type-limits]
  276 |         ASSERT(i >= 0 && i < m_size);
      |                ~~^~~~
/home/linus/Dev/git/serenity/AK/Vector.h:276:9: note: in expansion of macro ‘ASSERT’
  276 |         ASSERT(i >= 0 && i < m_size);
      |         ^~~~~~
/home/linus/Dev/git/serenity/AK/Vector.h: In instantiation of ‘T& AK::Vector<T, inline_capacity>::at(size_t) [with T = Parameter; long unsigned int inline_capacity = 0; size_t = long unsigned int]’:
/home/linus/Dev/git/serenity/AK/Vector.h:532:47:   required from ‘void AK::Vector<T, inline_capacity>::ensure_capacity(size_t) [with T = Parameter; long unsigned int inline_capacity = 0; size_t = long unsigned int]’
/home/linus/Dev/git/serenity/AK/Vector.h:518:9:   required from ‘void AK::Vector<T, inline_capacity>::grow_capacity(size_t) [with T = Parameter; long unsigned int inline_capacity = 0; size_t = long unsigned int]’
/home/linus/Dev/git/serenity/AK/Vector.h:462:9:   required from ‘void AK::Vector<T, inline_capacity>::append(T&&) [with T = Parameter; long unsigned int inline_capacity = 0]’
main.cpp:134:43:   required from here
/home/linus/Dev/git/serenity/AK/Vector.h:276:18: error: comparison of unsigned expression in ‘>= 0’ is always true [-Werror=type-limits]
  276 |         ASSERT(i >= 0 && i < m_size);
      |                ~~^~~~
/home/linus/Dev/git/serenity/AK/Vector.h:276:9: note: in expansion of macro ‘ASSERT’
  276 |         ASSERT(i >= 0 && i < m_size);
      |         ^~~~~~
cc1plus: note: unrecognized command-line option ‘-Wno-unknown-warning-option’ may have been intended to silence earlier diagnostics
cc1plus: all warnings being treated as errors
make[4]: *** [../../Makefile.common:111: main.host.o] Error 1
make[4]: Leaving directory '/home/linus/Dev/git/serenity/DevTools/IPCCompiler'
make[3]: *** [../../Makefile.common:137: /home/linus/Dev/git/serenity/DevTools/IPCCompiler/IPCCompiler] Error 2
make[3]: Leaving directory '/home/linus/Dev/git/serenity/Services/LaunchServer'
make[2]: *** [Makefile:10: ../../Services/LaunchServer/LaunchServerEndpoint.h] Error 2
make[2]: Leaving directory '/home/linus/Dev/git/serenity/Libraries/LibDesktop'
make[1]: *** [../Makefile.subdir:3: LibDesktop/] Error 2
make[1]: Leaving directory '/home/linus/Dev/git/serenity/Libraries'
make: *** [Makefile.subdir:3: Libraries] Error 2
make: Leaving directory '/home/linus/Dev/git/serenity'
```

--- a/AK/Vector.h
+++ b/AK/Vector.h
@@ -268,12 +268,12 @@ public:
 
     ALWAYS_INLINE const T& at(size_t i) const
     {
-        ASSERT(i >= 0 && i < m_size);
+        ASSERT(i < m_size);
         return data()[i];
     }
     ALWAYS_INLINE T& at(size_t i)
     {
-        ASSERT(i >= 0 && i < m_size);
+        ASSERT(i < m_size);
         return data()[i];
     }
 

